cosmic-update.jenkinsfile

```diff
@@ -0,0 +1,131 @@
+			}
+		}
+		// This stage backs up the gk_central database before it is modified.
+		stage('Setup: Back up gk_central - pre-update')
+		{
+			steps
+			{
+				script
+				{
+					withCredentials([usernamePassword(credentialsId: 'mySQLCuratorUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')])
+					{
+						utils.takeDatabaseDumpAndGzip("${env.GK_CENTRAL_DB}", "cosmic-update", "before", "${env.CURATOR_SERVER}")
+					}
+				}
+			}
+		}
+		// Builds the jar file using Maven.
+		stage('Setup: Build jar file')
+		{
+			steps
+			{
+				script
+				{
+					utils.buildJarFile()
+				}
+			}
+		}
+		// Download the data files. Done as a separate step from the main program's execution so that if
+		// the program fails, you don't get stuck waiting for a long download again. 
+		stage('Download COSMIC files')
+		{
+			steps
+			{
+				script
+				{
+					withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')])
+					{
+						sh "java -Xmx${env.JAVA_MEM_MAX}m -jar target/cosmic-update-*-jar-with-dependencies.jar -d PT72H -c $ConfigFile"
+					}
+				}
+			}
+		}
+		// Run the COSMIC Update
+		stage('Run COSMIC updates')
+		{
+			steps
+			{
+				script
+				{
+					withCredentials([file(credentialsId: 'Config', variable: 'ConfigFile')])
+					{
+						sh "java -Xmx${env.JAVA_MEM_MAX}m -jar target/cosmic-update-*-jar-with-dependencies.jar -u -c $ConfigFile"
+					}
+				}
+			}
+		}
+		// This stage backs up the gk_central database after modification.
+		stage('POST: Backup gk_central - post-update')
+		{
+			steps
+			{
+				script
+				{
+					withCredentials([usernamePassword(credentialsId: 'mySQLCuratorUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')])
+					{
+						utils.takeDatabaseDumpAndGzip("${env.GK_CENTRAL_DB}", "cosmic-update", "after", "${env.CURATOR_SERVER}")
+					}
+				}
+			}
+		}
+		// This stage archives the contents of the 'reports' folder generated by COSMIC Update and sends them in an email to the default recipients list.
+		stage('Post: Email COSMIC Update Reports')
+		{
+			steps
+			{
+				script
+				{
+					def releaseVersion = utils.getReleaseVersion()
+					sh "tar -czf cosmic-update-v${releaseVersion}-reports.tgz reports/"
+					emailext (
+						body: "Hello,\n\nThis is an automated message from Jenkins regarding an update for v${releaseVersion}. The COSMIC Update step has completed. Please review the reports attached to this email. If they look correct, these reports need to be uploaded to the Reactome Drive at Reactome>Release>Release QA>V${releaseVersion}_QA>V${releaseVersion}_COSMIC_Update_Reports. The URL to the new V${releaseVersion}_COSMIC_Update_Reports folder also needs to be updated at https://devwiki.reactome.org/index.php/Reports_Archive under 'COSMIC Update Reports'. Please add the current COSMIC report wiki URL to the 'Archived reports' section of the page. If the reports don't look correct, please email the developer running Release. \n\nThanks!",
```

Comment:
Not a huge deal, but could this be broken up into multi lines so it's easier to read in an IDE or text editor without a lot of horizontal scrolling?

Reply:
This was copied from Justin. I also think it's gross, I just assumed there was a reason he did that. Maybe Jenkinsfiles don't handle multi-line string expressions? I can look into it.

Reply:
Thanks!